internal/storage/badger/store.go

```diff
@@ -189,28 +189,31 @@ func (bdb *badgerDB) Close() error {
 	return nil
 }
 
-func (bdb *badgerDB) PutTTL(key []byte, value []byte, expireTS uint64) error {
-	defer bdb.opts.statsCli.Timing("badger.putTTL.latency.ms", time.Now())
-	err := bdb.db.Update(func(txn *badger.Txn) error {
-		kv := badger.NewEntry(key, value)
-		if expireTS > 0 {
-			kv.ExpiresAt = expireTS
-		}
-		return txn.SetEntry(kv)
-	})
-	if err != nil {
-		bdb.opts.statsCli.Incr("badger.putTTL.errors", 1)
+func (bdb *badgerDB) Put(pairs ...*storage.KVEntry) error {
+	metricsPrefix := "badger.put.multi"
+	if len(pairs) == 1 {
+		metricsPrefix = "badger.put.single"
 	}
-	return err
-}
+	defer bdb.opts.statsCli.Timing(metricsPrefix+".latency.ms", time.Now())
 
-func (bdb *badgerDB) Put(key []byte, value []byte) error {
-	defer bdb.opts.statsCli.Timing("badger.put.latency.ms", time.Now())
-	err := bdb.db.Update(func(txn *badger.Txn) error {
-		return txn.Set(key, value)
-	})
+	wb := bdb.db.NewWriteBatch()
+	defer wb.Cancel()
+	for _, kv := range pairs {
+		if kv == nil {
+			continue //skip nil entries
```

Comment:
why would it be nil ? should we avoid this check if unexpected ?

Reply:
A user can end up providing a sparse array. In that case, the code shouldn't throw NPE. This check is for that.